File: DotNetPrototypes/ExpenseCalculator/Department.cs
using System.Collections.Generic;

namespace ExpenseCalculator
{
    public class Department : IOrganizational
    {
        public string Name { get; set; }

        #region IOrganizational Members

        private List<IEmployee> employees = new List<IEmployee>();

        public List<IEmployee> Employees
        {
            get { return employees; }
        }

        public decimal GetTotalExpenses()
        {
            decimal expenses = 0;
            foreach (IEmployee employee in employees)
            {
                if (employee.GetType() == typeof(Manager))
                    expenses += ((Manager)employee).GetTotalExpenses();
                else
                    expenses += employee.ExpenseAllocation;
            }

            return expenses;
        }

        #endregion
    }
}

File: JavaPrototypes/ExpenseCalculator/src/com/example/expenseCalculator/Department.java
package com.example.expenseCalculator;

import java.util.ArrayList;
import java.util.List;

public class Department implements Organizational{
	
	private String name = "";
	private List<Employee> employees = new ArrayList<Employee>();
	
	public String getName()
	{
		return name;
	}
	
	public void setName(String value)
	{
		name = value;
	}

	public List<Employee> getEmployees() {
		return employees;
	}

	public double getTotalExpenses() {
		double expenses = 0;
		
		for(Employee employee : employees)
		{
			if (employee.getClass() ==  Manager.class)
				expenses += ((Manager)employee).getTotalExpenses();
			else
				expenses += employee.getExpenseAllocation();
		}
		
		return expenses;
	}

}
